monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -61,16 +61,30 @@ def _scan_addresses(
                 address = addresses.get_nowait()
                 logger.info(f"Scanning {address.ip}")
 
-                # TODO: Catch exceptions to prevent thread from crashing
-                ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
-                port_scan_data = self._puppet.scan_tcp_ports(address.ip, tcp_ports, tcp_timeout)
+                try:
+                    ping_scan_data = self._puppet.ping(address.ip, icmp_timeout)
+                except Exception as ex:
+                    logger.debug(f"Exception encountered when pinging {address.ip}: {str(ex)}")
+
+                try:
+                    port_scan_data = self._puppet.scan_tcp_ports(address.ip, tcp_ports, tcp_timeout)
+                except Exception as ex:
+                    logger.debug(
+                        f"Exception encountered when scanning TCP ports on {address.ip}: {str(ex)}"
+                    )
```

Comment:
If an exception is thrown, `ping_scan_data` or `port_scan_data` would be `None`. It's probably safer to return data structures with default values.

We can add a unit test for this case.